### DISCUSSION TITLE
How can I extend mapped properties?

### DISCUSSION THREAD

**Discussion by jbarranis** [nartc/mapper#339] [Q&A] [answered]
I have the following classes:

- Entity/EntityVm
- Customer/CustomerVm

I'm trying to map properties for both and join them together. I assume this is what Profiles can do. However, I get the this error on the line that calls `.createMap` inside the profile: `TypeError: Cannot read property '1' of undefined`

```
// entity.ts
export class Entity {
  @AutoMap()
  display_name: string;
  ...
}
export class EntityVm {
  @AutoMap()
  displayName: string;
  ...
}
export const entityProfile: MappingProfile = (mapper) => {
  mapper
    .createMap(Entity, EntityVm, {
      namingConventions: {
        source: new PascalCaseNamingConvention(),
        destination: new CamelCaseNamingConvention(),
      },
    })
    .forMember(
      (destination) => destination.displayName,
      mapFrom((source) => source.display_name)
    );
}

// customer.ts
export class Customer extends Entity {
   @Automap()
   custom_property: string;
}
export class CustomerVm extends EntityVm {
   @Automap()
   customProperty: string;
}
export customerProfile: MappingProfile = (mapper) => {
   // this doesn't cause an error, but I get warnings saying there are unmapped properties
   mapper.createMap(Customer, CustomerVm);
}
mapper.addProfile(entityProfile).addProfile(customerProfile);

```

Is this the proper way to extend mapped properties? I expect the output to be:
```
const mapped = mapper.map(data, Customer, CustomerVm);
// { displayName: "test", customProperty: "test" }
```

**Comment by micalevisk** [comment]
what if you change `PascalCaseNamingConvention` to `SnakeCaseNamingConvention`?

**Reply by jbarranis** [reply]
That doesn't resolve the issue. I feel like I may be implementing this incorrectly, which could be causing the downstream error. I updated my code above, which doesn't throw an error anymore (but it still doesn't do what I need).

**Comment by nartc** [comment, marked as answer]
@jbarranis hi, have you read this in the documentations? https://automapperts.netlify.app/docs/misc/limitations#property-name-with-number-or-special-characters I think this is one of the cases of limitations.

For your `Entity` and `EntityVm`, I think using `SnakeCaseNamingConvention` is more appropriate than `PascalCaseNamingConvention` given your example

**Reply by jbarranis** [reply]
hi @nartc Thanks. Yes I saw the special characters limitations documentation (also updated it to use `SnakeCaseNamingConvention`). I have a lot of fields in the classes, but my code example above just shows the one field. All of the fields don't seem to be mapped though.

Since I have `Customer extends Entity`, I want the mapper to automatically map properties I listed in `entity.ts`, but those don't seem to get picked up.

**Reply by nartc** [reply]
ah, have you tried? 

mapper.createMap(Customer, CustomerVm, { extends: [mapper.getMapping(Entity, EntityVm)] }

**Reply by jbarranis** [reply]
Ah ha! That's what I was missing. Thanks @nartc!
